### PR TITLE
some fixes

### PR DESCRIPTION
**Pull request by jakoch**
- added missing headers
- fixed some clang warnings/errors
  - added braces to fix precedence warning for "^" and "==" in f32x4_sse.hpp
  - fixed brace init
  - removed [nodiscard] from void functions ("function has no return")
  - removed [nodiscard] from friend function ("an attribute list cannot appear here")
- fixed std::overflow_error() trying to pass multiple arguments (instead of one) by using fmt:format()
- added "-stdlib=libc++" to compile options for Clang

### CODE REVIEW

File: src/ttauri/cast.hpp [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #include "assert.hpp"
 #include <type_traits>
 #include <concepts>
-#include <limits>
+#include <climits> // for identifier CHAR_BIT
```
Comment (takev) [inline src/ttauri/cast.hpp:12]: Lets not include comments for headers.

File: src/ttauri/datum.hpp [hunk 2]
```diff
@@ -28,6 +28,7 @@
 #include <numeric>
 #include <string_view>
 #include <cmath>
+#include <stdexcept> // for std::overflow_error
```
Comment (takev) [inline src/ttauri/datum.hpp:31]: no comment

File: src/ttauri/geometry/axis_aligned_rectangle.hpp [hunk 3]
```diff
@@ -418,7 +418,7 @@ class axis_aligned_rectangle {
      * @param rectangle The rectangle to fit inside the bounds.
      * @return A rectangle that fits inside the bounds
      */
-    [[nodiscard]] friend axis_aligned_rectangle
+    friend axis_aligned_rectangle
```
Comment (takev) [inline src/ttauri/geometry/axis_aligned_rectangle.hpp:421]: you should never discard the result of fit.

File: src/ttauri/geometry/numeric_array.hpp [hunk 4]
```diff
@@ -19,6 +19,7 @@
 #include <type_traits>
 #include <concepts>
 #include <bit>
+#include <climits> // for identifier CHAR_BIT
```
Comment (takev) [inline src/ttauri/geometry/numeric_array.hpp:22]: no comment

**Review by takev — CHANGES_REQUESTED** [review]

File: src/ttauri/geometry/axis_aligned_rectangle.hpp (hunk 3 above)
Comment (jakoch) [reply src/ttauri/geometry/axis_aligned_rectangle.hpp]: Re-added the `[[nodiscard]]`. This gives the following Clang error: https://godbolt.org/z/d546Pc

**Review by takev — APPROVED** [review]

**Pull request merged**